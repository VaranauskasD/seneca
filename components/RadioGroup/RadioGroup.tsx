import React from 'react'
import styled from 'styled-components'

import { Radio } from '../Radio'

interface RadioGroupProps {
  id: number
  options: { option: string; correct?: boolean }[]
  handleOption: (
    event: React.ChangeEvent<HTMLInputElement>,
    optionGroupId: number,
    correct: boolean
  ) => void
}

const StyledRadioGroup = styled.div<{ $quantity: number }>`
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  border: 2px solid #f9d29f;
  margin: 4px;
  text-align: center;
  text-decoration: none;
  border-radius: 16px;

  @media (min-width: ${(props) => `${props.theme.breakpoints.sm}px`}) {
    flex-direction: ${(props) => (props.$quantity > 2 ? 'column' : 'row')};
  }
`

export const RadioGroup = (props: RadioGroupProps) => {
  return (
    <React.Fragment>
      <StyledRadioGroup $quantity={props.options.length}>
        {props.options.map((option, key) => (
          <Radio
            key={`option-${props.id}-${key}`}
            id={key}
            optionGroupId={props.id}
            label={option.option}
            name={`option-${props.id}`}
            correct={option.correct || false}
            handleOption={props.handleOption}
          />
        ))}
      </StyledRadioGroup>
    </React.Fragment>
  )
}

export default RadioGroup
