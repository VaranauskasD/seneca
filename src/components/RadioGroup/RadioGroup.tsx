import React, { useEffect, useState } from 'react'
import styled from 'styled-components'

import { Radio } from '../Radio'

export interface RadioGroupProps {
  id: number
  options: { option: string; correct?: boolean }[]
  handleOption: (optionGroupId: number, correct: boolean) => void
  dynamicRatio: number
  isAnswered: boolean
}

const StyledRadioGroup = styled.div<{
  $quantity: number
  $dynamicRatio: number
}>`
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  border: 2px solid
    ${(props) =>
      `rgb(
        ${
          props.theme.colors.dynamic.secondary.start.r +
          props.theme.colors.dynamic.secondary.range.r * props.$dynamicRatio
        }, 
        ${
          props.theme.colors.dynamic.secondary.start.g +
          props.theme.colors.dynamic.secondary.range.g * props.$dynamicRatio
        }, 
        ${
          props.theme.colors.dynamic.secondary.start.b +
          props.theme.colors.dynamic.secondary.range.b * props.$dynamicRatio
        })`};
  margin: 4px;
  text-align: center;
  text-decoration: none;
  border-radius: 24px;

  @media (min-width: ${(props) => `${props.theme.breakpoints.sm}px`}) {
    flex-direction: ${(props) => (props.$quantity > 2 ? 'column' : 'row')};
    border-radius: ${(props) => (props.$quantity > 2 ? '24px' : '100px')};
  }

  :focus-within {
    outline: 2px solid ${(props) => props.theme.colors.primary};

    outline-offset: -2px;
  }
`

export const RadioGroup = (props: RadioGroupProps) => {
  const [selectedRadio, setSelectedRadio] = useState<number>(0)

  return (
    <StyledRadioGroup
      $quantity={props.options.length}
      $dynamicRatio={props.dynamicRatio}
    >
      {props.options.map((option, key) => (
        <Radio
          key={`option-${props.id}-${key}`}
          id={key}
          optionGroupId={props.id}
          label={option.option}
          name={`option-${props.id}`}
          checked={key === selectedRadio}
          correct={option.correct || false}
          handleCheck={setSelectedRadio}
          handleOption={props.handleOption}
          dynamicRatio={props.dynamicRatio}
          groupQuantity={props.options.length - 1}
          isAnswered={props.isAnswered}
        />
      ))}
    </StyledRadioGroup>
  )
}

export default RadioGroup
