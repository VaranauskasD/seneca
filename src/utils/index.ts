export { theme } from './theme'
export { customRender } from './test'
