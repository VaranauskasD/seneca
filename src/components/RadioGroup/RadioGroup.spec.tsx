import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { customRender as render } from '../../utils/test'
import { RadioGroup, RadioGroupProps } from './RadioGroup'

let mockRadioGroupProps: RadioGroupProps
beforeEach(() => {
  mockRadioGroupProps = {
    id: 0,
    options: [{ option: 'Cell wall', correct: true }, { option: 'Ribosomes' }],
    handleOption: () => {},
    dynamicRatio: 0,
    isAnswered: false,
  }
})

describe('RadioGroup', () => {
  it('renders correctly', async () => {
    const { asFragment } = render(<RadioGroup {...mockRadioGroupProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly with three radio options', async () => {
    mockRadioGroupProps.options = [
      { option: 'Cellulose' },
      { option: 'Mitochondria', correct: true },
      { option: 'Chroloplast' },
    ]
    const { asFragment } = render(<RadioGroup {...mockRadioGroupProps} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
