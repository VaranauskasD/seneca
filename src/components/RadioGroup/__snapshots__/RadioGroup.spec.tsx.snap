// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-hKMtZM fRXuCA"
  >
    <div
      class="sc-bczRLJ fNFKHi"
    >
      <label
        class="sc-dkzDqf hdgGtK"
      >
        <input
          checked=""
          class="sc-gsnTZi exMZOs"
          id="option-0-0"
          name="option-0"
          type="radio"
        />
        Cell wall
      </label>
    </div>
    <div
      class="sc-bczRLJ jNqizO"
    >
      <label
        class="sc-dkzDqf hdgGtK"
      >
        <input
          class="sc-gsnTZi exMZOs"
          id="option-0-1"
          name="option-0"
          type="radio"
        />
        Ribosomes
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`RadioGroup renders correctly with three radio options 1`] = `
<DocumentFragment>
  <div
    class="sc-hKMtZM efHAkz"
  >
    <div
      class="sc-bczRLJ bDYwXX"
    >
      <label
        class="sc-dkzDqf hdgGtK"
      >
        <input
          checked=""
          class="sc-gsnTZi exMZOs"
          id="option-0-0"
          name="option-0"
          type="radio"
        />
        Cellulose
      </label>
    </div>
    <div
      class="sc-bczRLJ hXUtLi"
    >
      <label
        class="sc-dkzDqf hdgGtK"
      >
        <input
          class="sc-gsnTZi exMZOs"
          id="option-0-1"
          name="option-0"
          type="radio"
        />
        Mitochondria
      </label>
    </div>
    <div
      class="sc-bczRLJ iGNBNP"
    >
      <label
        class="sc-dkzDqf hdgGtK"
      >
        <input
          class="sc-gsnTZi exMZOs"
          id="option-0-2"
          name="option-0"
          type="radio"
        />
        Chroloplast
      </label>
    </div>
  </div>
</DocumentFragment>
`;
